server/chat-plugins/mafia.ts

```diff
@@ -736,9 +736,15 @@ class Mafia extends Rooms.RoomGame<MafiaPlayer> {
 
 	distributeRoles() {
 		const roles = Utils.shuffle(this.roles.slice());
+		if (roles.length < this.players.length) {
+			throw new Chat.ErrorMessage(`Not enough roles for all players. Have ${roles.length} roles but ${this.players.length} players.`);
+		}
 		if (roles.length) {
 			for (const p of this.players) {
-				const role = roles.shift()!;
+				const role = roles.shift();
+				if (!role) {
+					throw new Chat.ErrorMessage(`Failed to distribute roles: ran out of roles.`);
+				}
```

Comment:
This shouldn't be a chat error; it leaves state inconsistent and should be prevented by the earlier check. So this should be a real error.

```suggestion
				if (!role) throw new Error(`Ran out of roles.`);
```